Mathlib/Analysis/Normed/Order/UpperLower.lean

```diff
@@ -154,3 +198,63 @@ theorem IsLowerSet.exists_subset_ball (hs : IsLowerSet s) (hx : x ∈ closure s)
 #align is_lower_set.exists_subset_ball IsLowerSet.exists_subset_ball
 
 end Fintype
+
+section Finite
+variable [Finite ι] {s t : Set (ι → ℝ)} {a₁ a₂ b₁ b₂ x y : ι → ℝ} {δ : ℝ}
+
+/-!
+#### Note
+
+The closure and frontier of an antichain might not be antichains. Take for example the union
+of the open segments from `(0, 2)` to `(1, 1)` and from `(2, 1)` to `(3, 0)`. `(1, 1)` and `(2, 1)`
+are comparable and both in the closure/frontier.
+-/
+
+protected lemma IsClosed.upperClosure (hs : IsClosed s) (hs' : BddBelow s) :
```

Comment:
Please add `pi` somewhere in the name (here and below). Also please add a TODO about finding a right generality for this lemma.